bin/sl-run.js

```diff
@@ -11,15 +11,42 @@ process.on('disconnect', function() {
 var config = require('../lib/config'); // May exit, depending on argv
 var log = config.logger;
 
-if(!config.profile) {
-  if(config.isMaster) {
-    log.error('supervisor running without StrongOps (unprofiled)');
-  }
-} else {
-  require('strong-agent').profile(undefined, undefined, {
-    quiet: config.isWorker, // Quiet in worker, to avoid repeated log messages
-    logger: config.logger,
-  });
+var agent = require('../lib/agent');
+var options = {
+  quiet: config.isWorker, // Quiet in worker, to avoid repeated log messages
+  logger: config.logger,
+};
+
+switch(config.profile) {
+  case false: // Profiling explicitly disabled.
+    if(config.isMaster) {
+      log.error('supervisor running without profiling');
+    }
+    break;
+
+  case undefined: // No explicit request for profiling or metrics.
+    // Start with StrongOps if app is registered. This will print warning
+    // messages to the console if the api key is not found, which is backwards
+    // compatible.
+    agent().profile(undefined, undefined, options);
+    // Otherwise, just start. This is a no-op if it is already started.
+    agent().start();
+    break;
+
+  case true: // Profiling or metrics explicitly enabled.
+    agent().configure(options);
+    // Only try to start StrongOps if they have registered, to avoid legacy
+    // warning messages. If an app is missing a name, profile may still fail
+    // to start, so drop-through to start(). We must re-supply options.
+    if (agent().config.key)
```

Comment:
Not required! We really need lint rules, I guess. Note the existing code base, some written by you, does not always use `{}` for single-statement if: https://github.com/strongloop/strong-supervisor/blob/868ba2ce8cdd2d407c8244a4d2455206953935da/lib/expander.js#L11-L18